libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -257,79 +214,27 @@ UARTDriver::device_type UARTDriver::_parseDevicePath(const char *arg)
         _flag = strdup(flag);
     }
 
-    if (strcmp(protocol, "udp") == 0) {
-        type = DEVICE_UDP;
-    } else if (strcmp(protocol, "udpin") == 0) {
-        type = DEVICE_UDPIN;
-    } else {
-        type = DEVICE_TCP;
-    }
-
-errout:
-
-    free(devstr);
-    return type;
-}
-
+    AP_HAL::OwnPtr<SerialDevice> device = nullptr;
 
-bool UARTDriver::_serial_start_connection()
-{
-    _device = new UARTDevice(device_path);
-    _connected = _device->open();
-    _device->set_blocking(false);
-    _flow_control = FLOW_CONTROL_DISABLE;
-
-    return true;
-}
-
-#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_QFLIGHT
-bool UARTDriver::_qflight_start_connection()
-{
-    _device = new QFLIGHTDevice(device_path);
-    _connected = _device->open();
-    _flow_control = FLOW_CONTROL_DISABLE;
-
-    return true;
-}
-#endif
-
-/*
-  start a UDP connection for the serial port
- */
-void UARTDriver::_udp_start_connection(void)
-{
-    bool bcast = (_flag && strcmp(_flag, "bcast") == 0);
-    _device = new UDPDevice(_ip, _base_port, bcast, false);
-    _connected = _device->open();
-    _device->set_blocking(false);
-
-    /* try to write on MAVLink packet boundaries if possible */
-    _packetise = true;
-}
+    if (strcmp(protocol, "udp") == 0 || strcmp(protocol, "udpin") == 0) {
+        bool bcast = (_flag && strcmp(_flag, "bcast") == 0);
+        _packetise = true;
+        if (strcmp(protocol, "udp") == 0) {
+            device = new UDPDevice(_ip, _base_port, bcast, false);
```

Comment:
Is there a convention we should use in order to distinguish OwnPtr from raw ones?